app/Models/Task.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Task extends Model
+{
+    use HasFactory;
+
+    protected $guarded = [];
+
+    protected $casts = [
+        'closed_at' => 'datetime',
+    ];
+
+    public function recipient()
+    {
+        return $this->belongsTo(User::class, 'recipient_user_id');
+    }
+
+    public function sender()
+    {
+        return $this->belongsTo(User::class, 'sender_user_id');
+    }
+
```

Comment:
suggestion: rename re: migration *and* add eager loading "capabilities".

```suggestion

    /**
     * The user which is the subject of the task.
     */
    public function subject()
    {
        return $this->belongsTo(User::class, 'reference_user_id');
    }

```